### PR TITLE
report: move renderer code to report/

### PR DESCRIPTION
**Pull request by connorjclark**
ref #12623, setup for #12689

Moving the renderer code from `lighthouse-core/report` to `report/`, and simplifying the directory structure. Better separation (core vs report), and provides room for us to continue to grow the report renderer (upcoming refactor + FR changes + maybe LHCI diff view?).

Renamed `report-template.html` to `standalone-template.html`, as I always confused it with `templates.html`. Also matches the `standalone` report terminology in the upcoming report renderer refactor.

~Need to follow with a strings import.~ This PR includes a strings import.

```
report
├── README.md
├── assets
│   ├── standalone-template.html
│   ├── styles.css
│   └── templates.html
├── renderer
│   ├── category-renderer.js
│   ├── crc-details-renderer.js
│   ├── details-renderer.js
│   ├── dom.js
│   ├── element-screenshot-renderer.js
│   ├── i18n.js
│   ├── logger.js
│   ├── performance-category-renderer.js
│   ├── psi.js
│   ├── pwa-category-renderer.js
│   ├── report-renderer.js
│   ├── report-ui-features.js
│   ├── snippet-renderer.js
│   ├── text-encoding.js
│   └── util.js
├── report-assets.js
├── report-generator.js
└── test
    ├── renderer
    │   ├── category-renderer-test.js
    │   ├── crc-details-renderer-test.js
    │   ├── details-renderer-test.js
    │   ├── dom-test.js
    │   ├── element-screenshot-renderer-test.js
    │   ├── i18n-test.js
    │   ├── performance-category-renderer-test.js
    │   ├── psi-test.js
    │   ├── pwa-category-renderer-test.js
    │   ├── report-renderer-test.js
    │   ├── report-ui-features-test.js
    │   ├── snippet-renderer-test.js
    │   ├── text-encoding-test.js
    │   └── util-test.js
    └── report-generator-test.js

4 directories, 36 files

```

### CODE REVIEW

File: lighthouse-report/test/renderer/category-renderer-test.js
Comment (connorjclark) [inline lighthouse-report/test/renderer/category-renderer-test.js:12]: minor tweak I did in these test files was dropping `fs` for `report-assets.js`

File: tsconfig.json
Comment (connorjclark) [inline tsconfig.json:27]: hmm, didn't mean to do this, but seems this line can be dropped?

File: lighthouse-report/test/proto-test.js
Comment (connorjclark) [inline lighthouse-report/test/proto-test.js:8]: see comment

File: lighthouse-core/scripts/i18n/collect-strings.js
Comment (connorjclark) [inline lighthouse-core/scripts/i18n/collect-strings.js:644]: idk why this changed

File: lighthouse-report/README.md
Comment (connorjclark) [inline lighthouse-report/README.md:1]: mostly just updated links here.

File: lighthouse-core/scripts/i18n/collect-strings.js
Comment (patrickhulce) [reply lighthouse-core/scripts/i18n/collect-strings.js:644]: that's somewhat concerning... :/

can we do a diff?

this is what I see on master

```
    "When an element doesn't have an accessible name, screen readers announce it with a generic name, making it unusable for users who rely on screen readers. $LINK_START_0$Learn more$LINK_END_0$.",
    "When an element doesn't have an accessible name, screen readers announce it with a generic name, making it unusable for users who rely on screen readers. $LINK_START_0$Learn more$LINK_END_0$.",
    'ARIA $MARKDOWN_SNIPPET_0$ elements have accessible names',
    'ARIA $MARKDOWN_SNIPPET_0$ elements have accessible names',
    'ARIA $MARKDOWN_SNIPPET_0$ elements do not have accessible names.',
    'ARIA $MARKDOWN_SNIPPET_0$ elements do not have accessible names.',
    "When an element doesn't have an accessible name, screen readers announce it with a generic name, making it unusable for users who rely on screen readers. $LINK_START_0$Learn more$LINK_END_0$.",
    'ARIA $MARKDOWN_SNIPPET_0$ elements have accessible names',
    'ARIA $MARKDOWN_SNIPPET_0$ elements do not have accessible names.',
    "When an element doesn't have an accessible name, screen readers announce it with a generic name, making it unusable for users who rely on screen readers. $LINK_START_0$Learn more$LINK_END_0$.",
    'ARIA $MARKDOWN_SNIPPET_0$ elements have accessible names',
    'ARIA $MARKDOWN_SNIPPET_0$ elements do not have accessible names.',
    "When an element doesn't have an accessible name, screen readers announce it with a generic name, making it unusable for users who rely on screen readers. $LINK_START_0$Learn more$LINK_END_0$.",
    '$MARKDOWN_SNIPPET_0$ elements have $MARKDOWN_SNIPPET_1$ text',
    '$MARKDOWN_SNIPPET_0$ elements have $MARKDOWN_SNIPPET_1$ text',
    '$MARKDOWN_SNIPPET_0$ elements do not have $MARKDOWN_SNIPPET_1$ text',
    '$MARKDOWN_SNIPPET_0$ elements do not have $MARKDOWN_SNIPPET_1$ text',
    'Document has a valid $MARKDOWN_SNIPPET_0$',
    'Document has a valid $MARKDOWN_SNIPPET_0$',
    'Potential Savings',
    'Potential Savings',
    'Failing Elements',
    'Failing Elements',
    'URL',
    'URL',
    'Consider uploading your GIF to a service which will make it available to embed as an HTML5 video.',
    'Consider uploading your GIF to a service which will make it available to embed as an HTML5 video.',
    'Consider using a $LINK_START_0$plugin$LINK_END_0$ or service that will automatically convert your uploaded images to the optimal formats.',
    'Consider using a $LINK_START_0$plugin$LINK_END_0$ or service that will automatically convert your uploaded images to the optimal formats.',
    'Consider uploading your GIF to a service which will make it available to embed as an HTML5 video.'
```

File: lighthouse-report/test/proto-test.js
Comment (patrickhulce) [reply lighthouse-report/test/proto-test.js:8]: my initial reaction was this doesn't belong in `lighthouse-report`, so I would say yes

**Review by patrickhulce — COMMENTED** [review]
thanks for splitting this up, I like the new directory structure 👍

File: lighthouse-core/scripts/i18n/collect-strings.js
Comment (connorjclark) [reply lighthouse-core/scripts/i18n/collect-strings.js:644]:
```
"When an element doesn't have an accessible name, screen readers announce it with a generic name, making it unusable for users who rely on screen readers. $LINK_START_0$Learn more$LINK_END_0$.",
'ARIA $MARKDOWN_SNIPPET_0$ elements have accessible names',
'ARIA $MARKDOWN_SNIPPET_0$ elements do not have accessible names.',
"When an element doesn't have an accessible name, screen readers announce it with a generic name, making it unusable for users who rely on screen readers. $LINK_START_0$Learn more$LINK_END_0$.",
'ARIA $MARKDOWN_SNIPPET_0$ elements have accessible names',
'ARIA $MARKDOWN_SNIPPET_0$ elements do not have accessible names.',
"When an element doesn't have an accessible name, screen readers announce it with a generic name, making it unusable for users who rely on screen readers. $LINK_START_0$Learn more$LINK_END_0$.",
'ARIA $MARKDOWN_SNIPPET_0$ elements have accessible names',
'ARIA $MARKDOWN_SNIPPET_0$ elements do not have accessible names.',
"When an element doesn't have an accessible name, screen readers announce it with a generic name, making it unusable for users who rely on screen readers. $LINK_START_0$Learn more$LINK_END_0$.",
'$MARKDOWN_SNIPPET_0$ elements have $MARKDOWN_SNIPPET_1$ text',
'$MARKDOWN_SNIPPET_0$ elements do not have $MARKDOWN_SNIPPET_1$ text',
'Document has a valid $MARKDOWN_SNIPPET_0$',
'Potential Savings',
'Failing Elements',
'Consider uploading your GIF to a service which will make it available to embed as an HTML5 video.',
'Consider using a $LINK_START_0$plugin$LINK_END_0$ or service that will automatically convert your uploaded images to the optimal formats.',
'Consider uploading your GIF to a service which will make it available to embed as an HTML5 video.'
```

diff https://www.diffchecker.com/TFvWBE9j

**Comment by connorjclark** [comment]
can we do a strings import w/o actually committing? I think so. there's a test that is failing b/c of missing renderer strings in en-XA

File: lighthouse-core/scripts/i18n/collect-strings.js
Comment (patrickhulce) [reply lighthouse-core/scripts/i18n/collect-strings.js:644]: hm, that's 10 more missing than there should be, I had to add `collisionStrings.push` where the first `collisions++` is

**Review by brendankenny — COMMENTED** [review]
are we seriously going to do another `lighthouse-*/` directory? :P

`report/`!

**Comment by connorjclark** [comment]
Haha I agree, but was gonna suggest renaming all of those in another PR and did lighthouse-report just to match. Would lighthouse-core to core be breaking?

Could also just make it report now. Wdyt

**Comment by brendankenny** [comment]
ha, I wasn't sure if anyone else would agree :) It probably makes sense to start with just `report/` and do it here, but not sure if anyone else cares

File: lighthouse-core/scripts/i18n/collect-strings.js
Comment (connorjclark) [reply lighthouse-core/scripts/i18n/collect-strings.js:644]: oh.. then:

image

Comment (connorjclark) [reply lighthouse-core/scripts/i18n/collect-strings.js:644]: `collectAllStringsInDir` does have some local data per directory, and renderer strings moved to a new folder, so that must be why. no problem here i guess ... except maybe in the "collision detection" code.

Comment (connorjclark) [reply lighthouse-core/scripts/i18n/collect-strings.js:644]: https://github.com/GoogleChrome/lighthouse/pull/12697

**Comment by connorjclark** [comment]
ok i'm doing it 
image

File: lighthouse-core/scripts/i18n/collect-strings.js
Comment (brendankenny) [reply lighthouse-core/scripts/i18n/collect-strings.js:644]: I didn't follow up after https://github.com/GoogleChrome/lighthouse/pull/12441/files#r630521367 :(

**Comment by connorjclark** [comment]
imported strings too

**Comment by connorjclark** [comment]
@paulirish can you turn off CDT tests for a bit? at least until the renderer changes land sometime next week. this PR renames `template.html` to `standalone-template.html`, which means some BUILD crap in CDT needs to change too, but we can't do it all at once. and then, many things need to change in BUILD for es modules to land. so best to disable for a bit.

**Comment by connorjclark** [comment]
@patrickhulce so we agree the string collision is not a real problem, and we will hash it out eventually (but no reason to block this PR). anything else?

**Review by patrickhulce — APPROVED** [review]

**Pull request merged**
